### PR TITLE
IAM-429-Automate approving and merging lib updates

### PR DESCRIPTION
**Pull request by bencekov**
update_libs_merge.yaml github action implements approving and merging charm lib updates. If merge requirements aren't met, auto-merge is enabled for said PR.

### CODE REVIEW

File: .github/workflows/update_libs_merge.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/checkout@v3
+      - name: Approve PR
+        run: |
+          gh pr review --approve || true
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}
+      - name: Merge PR
+        # If pr can't be merged because requirements aren't met automerge is enabled
+        run: |
+          gh pr merge --auto --body "Update charm libs" --subject "(Automated) Update Charm Libs" || true
```
Comment (shipperizer) [inline .github/workflows/update_libs_merge.yaml]: You might want to move this part in here https://github.com/canonical/identity-platform-login-ui-operator/blob/main/.github/workflows/update_libs.yaml basically enable automerge on pr creation

File: .github/workflows/update_libs_merge.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# reusable workflow for automatically approving PRs updating charm libraries
+name: Approve and Merge Updating Charm Libraries
+run-name: CI for approving and merging automated-update-charm-libs PRs
+
+on:
+  push:
+    branches:
+      - "automated-update-charm-libs"
```
Comment (shipperizer) [inline .github/workflows/update_libs_merge.yaml]: Have a look at the pr https://github.com/canonical/identity-platform-login-ui-operator/pull/59

adding the branch in there and then making sure that we have a branch protection rule set (check the settings for the renovate/* pattern) will do the job without an extra workflow

File: .github/workflows/update_libs_approve.yaml
Comment (nsklikas) [inline .github/workflows/update_libs_approve.yaml:1]: This should be the same file as https://github.com/canonical/identity-platform-login-ui-operator/blob/main/.github/workflows/renovate-approve.yaml.

Just add the branch `automated-update-charm-libs` there

Comment (bencekov) [reply .github/workflows/update_libs_approve.yaml:1]: sure

**Review by nsklikas — APPROVED** [review]

**Pull request merged**